### PR TITLE
[Docs] Connect Sidecar Proxies: Fixes 'must may' typo and adds formatting to follow other docs

### PR DESCRIPTION
**Pull request by johncowen**
1. Corrects a `must may` typo - replaced with `should` (??).
2. Adds formatting and `<required>` to follow the styling of other docs, removes 'optional' which we seem to just use empty strings for elsewhere.
3. Replaces instances of `string` with `int` in certain places (ports)

Lastly I wasn't sure whether to remove the optional thing here, but I looked elsewhere and the only place we do that is in `operator/area`. Happy to revert that if we still want that here. I think the main thing here is the `must may` typo.

### CODE REVIEW

File: website/source/docs/connect/proxies.html.md [hunk 1]
```diff
@@ -113,33 +113,33 @@ until at least 1.5.0 but it's highly recommended to switch to using
...
    instance is a side-car for which application instance and will enable
    fine-grained analysis of the metrics coming from the proxy.
 
- - `local_service_address` `string: <optional>` - Specifies the address a side-car
+ - `local_service_address` `(string: "")` - Specifies the address a side-car
    proxy should attempt to connect to the local application instance on.
    Defaults to 127.0.0.1.
 
- - `local_service_port` `int: <optional>` - Specifies the port a side-car
+ - `local_service_port` `(int: <required>)` - Specifies the port a side-car
```
Comment (judithpatudith) [inline website/source/docs/connect/proxies.html.md]: I think this is the only one that switched from optional to required; just want to make sure it isn't a mistake.

Comment (banks) [reply website/source/docs/connect/proxies.html.md]: Yeah it is optional. This is different form the upstream bind port above which is required currently.

If not provided we default to the service's advertised port as noted in the comment.

Comment (johncowen) [reply website/source/docs/connect/proxies.html.md]: Ah glad you queried that @judithpatudith ! Can't think why I thought it needed changing, does this effect line 81 above also? Will look back at what I was doing also.

I _think_ I might have thought 'this should be optional' as I don't think normal proxies require it, but sidecars do? Is that the difference here?

Comment (johncowen) [reply website/source/docs/connect/proxies.html.md]: Cool, I've reverted that `required` back to `optional` here

**Pull request merged**